pkg/cmd/grafana-server/main.go

```diff
@@ -56,16 +58,56 @@ func main() {
 		os.Exit(0)
 	}
 
-	if *profile {
+	ultimateProfileEnabled := *profile
+	ultimateProfilePort := *profilePort
+	profileEnv, exists := os.LookupEnv("GF_PROCESS_PROFILE")
+	if exists {
+		enabled, parseErr := strconv.ParseBool(profileEnv)
+		if parseErr != nil {
+			panic(parseErr)
+		} else {
+			ultimateProfileEnabled = enabled
+			profilePortEnv, exists := os.LookupEnv("GF_PROCESS_PROFILE_PORT")
+			if exists {
+				port, parseErr := strconv.ParseInt(profilePortEnv, 0, 64)
+				if parseErr != nil {
+					panic(parseErr)
+				} else {
+					ultimateProfilePort = int(port)
+				}
+			}
+		}
+	}
+
+	ultimateTracingEnabled := *tracing
+	tracingEnv, exists := os.LookupEnv("GF_PROCESS_TRACING")
+	if exists {
+		enabled, parseErr := strconv.ParseBool(tracingEnv)
+		if parseErr != nil {
+			panic(parseErr)
+		} else {
+			ultimateTracingEnabled = enabled
+		}
+	}
+
+	if ultimateProfileEnabled {
 		runtime.SetBlockProfileRate(1)
 		go func() {
-			err := http.ListenAndServe(fmt.Sprintf("localhost:%d", *profilePort), nil)
+			err := http.ListenAndServe(fmt.Sprintf("localhost:%d", ultimateProfilePort), nil)
 			if err != nil {
 				panic(err)
 			}
 		}()
+	}
+
+	if ultimateTracingEnabled {
+		ultimateTracingFile := *tracingFile
+		tracingFileEnv, exists := os.LookupEnv("GF_PROCESS_TRACING_OUTPUT")
```

Comment:
Could you use `os.Getenv` and verify that the value isn't empty instead? Now it could still be the case that you get an empty value.